File: backend/src/main/java/com/hotelbooking/service/HotelService.java
package com.hotelbooking.service;

import com.hotelbooking.database.HotelCategory;
import com.hotelbooking.dto.RoomResponse;
import com.hotelbooking.dto.hotel.CityResponse;
import com.hotelbooking.dto.hotel.HotelDetailResponse;
import com.hotelbooking.dto.hotel.HotelSummaryResponse;
import com.hotelbooking.dto.hotel.HotelUpsertRequest;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.math.BigDecimal;
import java.util.List;

public interface HotelService {

    Page<HotelSummaryResponse> search(
            String citySlug,
            String city,
            String name,
            HotelCategory category,
            Integer minStars,
            BigDecimal minRating,
            BigDecimal minPrice,
            BigDecimal maxPrice,
            Boolean breakfastIncluded,
            Boolean freeCancellation,
            Boolean petFriendly,
            List<String> amenityCodes,
            Pageable pageable
    );

    HotelDetailResponse getBySlug(String slug);

    HotelDetailResponse getById(Long id);

    List<HotelSummaryResponse> featured();

    List<CityResponse> popularCities();

    List<CityResponse> listTelanganaCities();

    HotelDetailResponse submitForApproval(HotelUpsertRequest request, Long ownerUserId);

    HotelDetailResponse approve(Long hotelId);

    HotelDetailResponse reject(Long hotelId, String reason);

    List<HotelSummaryResponse> listPendingApproval();

    List<HotelSummaryResponse> listOwned(Long ownerUserId);

    List<RoomResponse> listRoomsBySlug(String slug);
}

File: frontend/src/services/hotelService.js
import api from './api';

export const hotelService = {
  search(params = {}) {
    return api.get('/hotels/search', { params }).then((res) => res.data);
  },
  featured() {
    return api.get('/hotels/featured').then((res) => res.data);
  },
  getBySlug(slug) {
    return api.get(`/hotels/${slug}`).then((res) => res.data);
  },
  rooms(slug) {
    return api.get(`/hotels/${slug}/rooms`).then((res) => res.data);
  },
};

export const cityService = {
  list() {
    return api.get('/cities').then((res) => res.data);
  },
  popular() {
    return api.get('/cities/popular').then((res) => res.data);
  },
};
